librz/core/disasm.c

```diff
@@ -2197,7 +2209,7 @@ static void ds_show_flags(RzDisasmState *ds, bool overlapped) {
 	RzAnalysisBlock *switch_block = NULL;
 	const char *switch_enum_name = NULL;
 	rz_list_foreach (uniqlist, iter, flag) {
-		if (!overlapped && f && f->addr == flag->offset && !strcmp(flag->name, f->name)) {
+		if (!overlapped && f && f->addr == flag->offset && (!strcmp(flag->name, f->name) || is_sym_dbg_equal(flag->name, f->name))) {
```

Comment:
Can you extract such a long condition into a separate static inline function to simplify the code?